interfaces/polkit/spec.go

```diff
@@ -0,0 +1,106 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package polkit
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/snap"
+)
+
+type Policy []byte
+
+// Specification keeps all the polkit policies.
+type Specification struct {
+	policyFiles map[string]Policy
+}
+
+// AddPolicy adds a polkit policy file to install.
+func (spec *Specification) AddPolicy(name string, content Policy) error {
+	if old, ok := spec.policyFiles[name]; ok && !bytes.Equal(old, content) {
+		return fmt.Errorf("interface %q requires conflicting polkit policy content", name)
```

Comment:
so it's a bit of guess here to prefix this error with "interface", will name be an interface name?

Reply:
In PR #10219, I've used the plug name. Comparing with how the systemd backend is uses in the gpio and pwm interfaces, perhaps it should prefix with the interface name too.

Reply:
what to do here is clarified within the changes to the systemd backend

Reply:
this needs a bit more work to match the detail level conveyed in the corresponding systemd/spec.go code, see the code there in master

Reply:
I've adjusted the error message a bit, but including the policy file's content is not really appropriate: this is often kilobytes of data.